controllers/chat/getSignedChannelList.js

```diff
@@ -82,7 +122,17 @@ const getSignedChannelList = async (req, res) => {
       queriedChannels.push(...response);
     });
 
-    const channels = queriedChannels.reduce(__filterAndTransformChannelData, []);
+    // TODO: Change to this channel filtering after admin system has implemented channel image saving;
+    // const channels = queriedChannels.reduce(__filterAndTransformChannelData, []);
+
+    const channels = [];
+    for (let i = 0; i < queriedChannels.length; i++) {
+      const _channelItem = queriedChannels[i];
+      const _filteredChannel = await __filterAndTransformChannelData(_channelItem);
+
+      if (_filteredChannel) channels.push(_filteredChannel);
+    }
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The change from `reduce` to a plain loop is necessary due to the asynchronous nature of the `__filterAndTransformChannelData` function. However, this could potentially slow down the execution as it waits for each iteration to complete before moving on to the next one. If the number of channels is large, this could lead to performance issues. Consider using `Promise.all` with `map` to handle all promises concurrently.

```diff
    // TODO: Change to this channel filtering after admin system has implemented channel image saving;
    // const channels = queriedChannels.reduce(__filterAndTransformChannelData, []);

-     const channels = [];
-     for (let i = 0; i < queriedChannels.length; i++) {
-       const _channelItem = queriedChannels[i];
-       const _filteredChannel = await __filterAndTransformChannelData(_channelItem);
- 
-       if (_filteredChannel) channels.push(_filteredChannel);
-     }
+     const channelsPromises = queriedChannels.map(_channelItem => __filterAndTransformChannelData(_channelItem));
+     const channels = (await Promise.all(channelsPromises)).filter(Boolean);

     return res.status(200).json(responseSuccess('Success retrieve channels', channels));
   } catch (error) {
     return res.status(error.statusCode ?? error.status ?? 400).json({
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->